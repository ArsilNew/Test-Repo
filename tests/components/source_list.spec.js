import React from "react";
import { shallow } from "./Utils";

import WithSourceList, { SourceList } from "../../src/components/source_list";

const CustomComponent = jest
  .fn(() => <div>Custom Component</div>)
  .mockName("mockedComponent");

const custom_messages = {
  searchPlaceholder: "Find...",
  noItemsMessage: "No entries available...",
  noneSelectedMessage: "Nothing",
  selectedMessage: "Checked",
  selectAllMessage: "Check all",
  clearAllMessage: "Uncheck all",
};

const selectAllItems = jest.fn().mockName("selectAllItems");
const filterItems = jest.fn().mockName("filterItems");
const selectItem = jest.fn().mockName("selectItem");

describe("SourceList", () => {
  test("default snapshot", () => {
    const { tree } = shallow(<SourceList />);
    expect(tree).toMatchSnapshot();
  });

  test("custom selectAllRenderer", () => {
    const { tree } = shallow(
      <SourceList selectAllRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom searchRenderer", () => {
    const { tree } = shallow(<SourceList searchRenderer={CustomComponent} />);
    expect(tree).toMatchSnapshot();
  });

  test("custom noItemsRenderer", () => {
    const { tree } = shallow(<SourceList noItemsRenderer={CustomComponent} />);
    expect(tree).toMatchSnapshot();
  });

  test("custom itemRenderer", () => {
    const { tree } = shallow(<SourceList itemRenderer={CustomComponent} />);
    expect(tree).toMatchSnapshot();
  });

  test("custom messages", () => {
    const { tree } = shallow(<SourceList messages={custom_messages} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedIds", () => {
    const { tree } = shallow(<SourceList selectedIds={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems", () => {
    const { tree } = shallow(<SourceList selectedItems={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectAllItems", () => {
    const { tree } = shallow(<SourceList selectAllItems={selectAllItems} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed filterItems", () => {
    const { tree } = shallow(<SourceList selectAllItems={filterItems} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectItem", () => {
    const { tree } = shallow(<SourceList selectAllItems={selectItem} />);
    expect(tree).toMatchSnapshot();
  });

  test("can remove select all", () => {
    const { tree } = shallow(<SourceList showSelectAll={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("can remove search", () => {
    const { tree } = shallow(<SourceList showSearch={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("show max selection tooltip", () => {
    const { tree } = shallow(
      <SourceList
        messages={{ disabledItemsTooltip: "You can select up to 4 items" }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  test("will pass itemHeight", () => {
    const { tree } = shallow(<SourceList itemHeight={10} />);
    expect(tree).toMatchSnapshot();
  });

  test("will pass selectAllHeight", () => {
    const { tree } = shallow(
      <SourceList itemHeight={10} selectAllHeight={60} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("will pass selectAllHeight without itemHeight", () => {
    const { tree } = shallow(<SourceList selectAllHeight={60} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed filteredItems with group", () => {
    const { tree } = shallow(
      <SourceList
        filteredItems={[
          { id: 1, label: "item1", group: "group1" },
          { id: 2, label: "item2", group: "group2" },
          { id: 3, label: "item3", group: "group1" },
        ]}
        withGrouping
      />
    );
    expect(tree).toMatchSnapshot();
  });
  test("passed source_list wrapped up to HOC withSearch", () => {
    const { tree } = shallow(
      <WithSourceList
        selectAllHeight={60}
        itemHeight={10}
        filterItems={filterItems}
        showSearch={true}
        filteredItems={[{ id: 1, label: "item1", group: "group1" }]}
        searchValue="1"
        selectAllItems={selectAllItems}
        withGrouping
        selectedIds={[1]}
        selectedItems={[1]}
        itemRenderer={CustomComponent}
        messages={custom_messages}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
