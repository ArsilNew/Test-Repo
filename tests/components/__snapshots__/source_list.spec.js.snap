// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceList can remove search 1`] = `
<Column>
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList can remove select all 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList custom itemRenderer 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[MockFunction mockedComponent]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[MockFunction mockedComponent]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList custom messages 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Find..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Check all"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage="No entries available..."
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList custom noItemsRenderer 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[MockFunction mockedComponent]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList custom searchRenderer 1`] = `
<Column>
  <mockConstructor
    onChange={undefined}
    searchIcon={undefined}
    searchPlaceholder={undefined}
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList custom selectAllRenderer 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <mockConstructor
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage={undefined}
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList default snapshot 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList passed filterItems 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={[MockFunction filterItems]}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList passed filteredItems with group 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={
      Array [
        Object {
          "disabled": true,
          "id": "group1",
          "isGroup": true,
          "label": "group1",
        },
        Object {
          "group": "group1",
          "id": 1,
          "label": "item1",
        },
        Object {
          "group": "group1",
          "id": 3,
          "label": "item3",
        },
        Object {
          "disabled": true,
          "id": "group2",
          "isGroup": true,
          "label": "group2",
        },
        Object {
          "group": "group2",
          "id": 2,
          "label": "item2",
        },
      ]
    }
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList passed selectAllItems 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={[MockFunction selectAllItems]}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList passed selectItem 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={[MockFunction selectItem]}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList passed selectedIds 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={
      Array [
        1,
        2,
      ]
    }
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={
      Array [
        1,
        2,
      ]
    }
  />
</Column>
`;

exports[`SourceList passed selectedItems 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList show max selection tooltip 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={40}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip="You can select up to 4 items"
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList will pass itemHeight 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={10}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={10}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList will pass selectAllHeight 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={60}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={10}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`SourceList will pass selectAllHeight without itemHeight 1`] = `
<Column>
  <Search
    onChange={undefined}
    searchIcon={[Function]}
    searchPlaceholder="Search..."
    value={undefined}
  />
  <SelectAll
    height={60}
    isAllSelected={false}
    onClick={undefined}
    renderer={[Function]}
    selectAllMessage="Select All"
    selectedIds={Array []}
  />
  <ItemsList
    disabled={false}
    disabledItemsTooltip={undefined}
    height={400}
    itemHeight={40}
    items={Array []}
    listRenderer={[Function]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={1}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;
